ballerina/tests/test.bal

```diff
@@ -0,0 +1,297 @@
+        "ids": [batchTicketId1, batchTicketId2],
+        "properties": [],
+        "inputs": []};
+    BatchResponseSimplePublicObject|BatchResponseSimplePublicObjectWithErrors|error response8 = check HubSpotClient->/batch/read.post(payload);
+    if (response8 is BatchResponseSimplePublicObject|BatchResponseSimplePublicObjectWithErrors) {
+        test:assertTrue(response8.status == "COMPLETE", "Response is not a valid");
+    } else {
+        test:assertFail("Response is not valid");
+    }
+}
+
+// Test to update a batch of tickets by id
+@test:Config{
+    groups: ["live_tests", "batch_tests"],
+    dependsOn: [createBatchTickets]
+}
+function updateBatchTickets() returns error? {
+    BatchInputSimplePublicObjectBatchInput payload = {
+        "inputs": [
+            {
+                "id": batchTicketId1,
+                "properties": {
+                    "subject": "Updated Troubleshoot Report 1",
+                    "hs_ticket_priority": "MEDIUM",
+                    "hs_pipeline": "0"
+                }
+            },
+            {
+                "id": batchTicketId2,
+                "properties": {
+                    "subject": "Updated Troubleshoot Report 2",
+                    "hs_ticket_priority": "LOW",
+                    "hs_pipeline": "0"
+                }
+            }
+        ]
+    };
+    BatchResponseSimplePublicObject|BatchResponseSimplePublicObjectWithErrors|error response9 = check HubSpotClient->/batch/update.post(payload);
+    test:assertTrue(response9 is BatchResponseSimplePublicObject|BatchResponseSimplePublicObjectWithErrors, "Error in updating the tickets");
+}
+
+// Create or update a batch of tickets by unique property values
+@test:Config{
+    groups: ["live_test1"]
+}
+isolated function createOrUpdateBatchTicketsByUniquePropertyValues() returns error? {
+    BatchInputSimplePublicObjectBatchInputUpsert payload = {
+        "inputs": [
+            {
+                "idProperty": "string",
+                "objectWriteTraceId": "string",
+                "id": "18189789866",
+                "properties": {
+                    "additionalprop1": "string",
+                    "additionalprop2": "string",
+                    "additionalprop3": "string"
+                }
+            }
+        ]
+    };
+    BatchResponseSimplePublicObject|BatchResponseSimplePublicObjectWithErrors|error response10 = check HubSpotClient->/batch/upsert.post(payload);
+    test:assertTrue(response10 is BatchResponseSimplePublicObject, "Error in creating or updating the tickets");
+    return;
+}
+
+// Merge two tickets of same type
+@test:Config{
+    groups: ["live_tests"],
+    dependsOn: [createTicket, createBatchTickets]
+}
+function mergeTickets() returns error? {
+    PublicMergeInput payload = {
+        "objectIdToMerge": ticketId,
+        "primaryObjectId": batchTicketId1
+    };
+    SimplePublicObject|error response11 = check HubSpotClient->/merge.post(payload);
```

Comment:
When you are using the `check` expression, you don't have to have `error` in the expected type. Check similar places.

```suggestion
    SimplePublicObject response11 = check HubSpotClient->/merge.post(payload);
```